---
title: 'Chapter 4: Population Structure and Gene Flow'
author: "Simon Renny-Byfield"
date: "November 3, 2014"
output: html_document
---

This document constitutes my personal notes based on "Population Genetics" by Matthew. B. Hamilton First Ed. 2009. They can be used, modified and distrubted freely under the terms of the GNU, provided the author of the document (myself) and the author of the book on which the notes are based (Matthew B. Hamilton), are credited.  
    
<br>    

***Genetic Populations***

* **Panmixia** The notion that a population is a single colection of individauls where mating is random and movement of individuals is uniform $\longrightarrow$ this is rarely true for natural populations.

Several things can cause devaition from *Panmixis*. Movement of individuals is often not random and the chance of two individuals mating often depends on their position in the population $\longrightarrow$ this is known as **Population Structure**. This harks back the the earlier chapter where Hamilton discusses **Isolation by distance**.

But what are the impacts of popultion structure on local allele and genetype frequencies?

Isolation can be thought of as a form of **Inbreeding** and results in a reduction in effective population size.

Hamilton points of the difference between *migration* and *gene flow*. Migration is simply the movement of individuals from one place to another, which may or may not result in gene flow. Gene flow is the movement of alleles between populations. Also gene flow can occur without any migration (in the case of long distance pollen dispersal for example).

<br>  

***Direct measures of gene flow***

You can, using genetic markers, identify the parents of a set pf progeny, and therefore reconstruct the matings that occured between the parents.

Hamilton discusses parentage tests (usually the female parent is known ad you are estimating the likely father). Typically genotyping at highly variable micro-satellite markers gives a good indication of the likely paternal line. A caveat is that you need to know the likely hood of having the same haplotype by chance. For example, the odds of having the same allele at a given locus is the odds of the progeny being homozygous or heterozygote for that allele at the given locus.

$$P(random\space match) = p^{2}_{i} + 2p_{i}(1-p_{i})$$

**under the assumption of random mating and panmixia**. Assuming that all loci are indepednant you can multiply the probabilities across all loci considered to give you the chance of a random match.

$$\prod^{loci}_{i=1} (p^{2}_{i} + 2p_{i}(1-p_{i}))$$

Random matches become less likely as scored allele are lower in frequency **AND** the number of loci under consideration increases.

Additionally the **Exclusion probability** is simply 1 minus the probabiltiy of matching by chance:

$$p(exclusion) = 1 - P(random\space match)$$

For a sample of $n$ individuals the total probability of exclusion (in this case excluding ALL the potential parents) is 

$$P(exclusion\space for\space n\space individuals) = (1-P(random\space match))^{n}$$

Thus the chance of exclusion decreases as you sample more individuals. This is the same thing as saying the probability of matching someone at random increases as you sample more individuals. From this we can calculate the chances that a random match *does* occur.

$$ = 1-P(exclusion\space for\space n\space individuals)$$

or

$$1 - (1-P(random\space match))^{n}$$ 

Importantly this probability can be thought of as the likelyhood that the **incorrect parent is assigned after the genotype test $\longrightarrow$ this is often refferred to as **cryptic gene flow**. This essentailly boils down to the fact that you cannot assertain the parent of one of the offspring and so the correct genotype has not been samples. It is missing for one reason or another. This could be :

1.  The parent may not be present in the sample (i.e. gene flow, death, migration??)
2.  The gene arises from outsode the bounds of your study plot (off site gene flow).

The are **estimates** of direct gene flow!!


***Fixation indices to measure the pattern of population subdivision***

One can assess population sctructure by taking samples from geographical locations and examining genotype. You can compare the information to what you would expect if the were no population structure.

For example the Fixation index can be expanded to cover deviations from the expected numbers of heterozygotes to descibe:

* 1 devaition due to non-randome mating within a population.
* 2 devaition due to population structure.

Remember that heterozygosity is $H= 1- F$ (that is 1 minus the homozygosity) and can be related to the fixation index $F$.

It is possible to calculate heterozygosity at three levels of population structure:

[1] $$H_{I}=\frac{1}{n}\sum^{n}_{i=1}\hat{H}_{i}$$

This is the average observed $H$ within each of the $n$ subpopulations.

[2] $$H_{S}=\frac{1}{n}\sum^{n}_{i=1}2p_{i}q_{i}$$

This is the average expected $H$ within each subpopulation, assuming random mating.

[3] $$H_{T}=2\bar{p}\bar{q}$$

This is expected $H$ within of the entire population, assuming random mating where $\bar{p}$ and $\bar{q}$ are the average allele frequencies across all populations.

But what if these sub-populations are different sizes. Say sub-pop1 is twice the size of sub-pop2, the simple avergae is not suitable surely...

**Various measures of fixation index**

| Index        | Meaning           | 
| ------------- |:-------------:| 
| $F_{IS}=\frac{H_{S}-\hat{H}_{I}}{H_{S}}$  | The mean difference between obs and HWE exp $H$  within each sub-pop|
| $F_{ST}=\frac{H_{T}-{H}_{S}}{H_{T}}$      | The reduction in $H$ due to allele fequency divergence between pops      | 
| $F_{IT}=\frac{H_{T}-{H}_{I}}{H_{T}}$ | The heterozygosity within a sub-pops given non-random mating and divergence between sub-pops      |

When two sub-populations are in complete panmixia (i.e. they are not really sub-populations) $H_{T}$ and $H_{S}$ are identicle and $F_{ST}$ is 0 (i.e. no population sub-division).

Another way to *estiamate* $F_{ST}$ is to measure the varience in allele frequency between populations. The varience amoung $n$ populations can be given as:

* $$var(p)=\frac{1}{n}\sum_{i=1}^{n}(p_{i} - \bar{p})^2$$

Which then leads to:

* $$F_{ST} = \frac{var(p)}{\bar{p}\bar{q}}$$

This is often used to estimate $F_{ST}$ ($\hat{F_{ST}}$) using real genotype data. Importantly, sub-popultions divergence in allele frequenciesdue to drift so it is crucial to calculate $F_{ST}$ over many many loci across the genome. 

##Population sub-division and the Wahlund effect##


